### PR TITLE
update performance_summarizer.py

### PR DESCRIPTION
**Pull request by Yu-Zhewen**
- fix print error for num of runs >= 10
- print average time (provisional, see https://github.com/nod-ai/iree-amd-aie/pull/931#discussion_r1858179653)

### CODE REVIEW

**Comment by jtuyls** [comment]
@Yu-Zhewen As a follow-up on my comment on how to average across runs without AIE going to sleep (https://github.com/nod-ai/iree-amd-aie/pull/931#discussion_r1858179653), I created this PR: https://github.com/nod-ai/iree-amd-aie/pull/936. This should get you to a similar result as this one, but it should be more accurate.

**Comment by newling** [comment]
> fix print error for num of runs >= 10
Haha I see the bug now. "#Run 1" and "#Run 10" both match the "check for first run" check. Thanks for fixing.

**Comment by newling** [comment]
Nit: please make the titles of PRs as descriptive as possible. So something like "[CI] Additionally print mean time of runs in performance_summarizer"

**Comment by newling** [comment]
{6C270AE1-11A2-404D-AC9C-51F731EDFCEA}

I guess the new line doesn't need to mention the test name again?

File: build_tools/ci/cpu_comparison/performance_summarizer.py [hunk 1]
```diff
@@ -13,15 +13,28 @@
...
-            if not first_print:
-                print("\n" + line.split()[-1])
-            else:
-                print(line.split()[-1])
-            first_print = False
-        if "IREE_AMDAIE" in line:
+        if "Run #1 of" in line:
+            if kernel_times:
+                print(
+                    f"Average kernel time for {test_name}: {sum(kernel_times) / len(kernel_times):.4f} ms"
```
Comment (newling) [inline build_tools/ci/cpu_comparison/performance_summarizer.py:25]:
```suggestion
                    f"Mean kernel time: {sum(kernel_times) / len(kernel_times):.4f} [ms]"
```

File: build_tools/ci/cpu_comparison/performance_summarizer.py [hunk 2]
```diff
@@ -13,15 +13,28 @@
...
+            print(f"\n{test_name}")
+        elif "IREE_AMDAIE" in line:
             print(line.strip())
+            kernel_times.append(
+                float(line.split("Kernel time:")[1].split("[ms]")[0].strip())
+            )
+
+    if kernel_times:
+        print(
+            f"Average kernel time for {test_name}: {sum(kernel_times) / len(kernel_times):.4f} ms"
```
Comment (newling) [inline build_tools/ci/cpu_comparison/performance_summarizer.py:38]:
```suggestion
            f"Mean kernel time: {sum(kernel_times) / len(kernel_times):.4f} [ms]"
```

**Review by newling — APPROVED** [review]
Looks good, just the one suggestion about the new line printed. Accepting in advance because it's good as is.

**Comment by Yu-Zhewen** [comment]
closing this without a merge, since we have a better solution to calculate the average execution time #936

**Pull request closed without merging**